File: superHotelFront/src/app/models/hotel.ts
import { City } from './city';
import { Hotelier } from './hotelier';

export class Hotel {
  id: number;
  name: string;
  address: string;
  email: string;
  phone: string;
  image: string;
  numberOfRooms: number;
  numberOfStars: number;
  averagePrice:number;
  city: City;
  hotelier: Hotelier;

  constructor(
    id: number,
    name: string,
    address: string,
    email: string,
    phone: string,
    image: string,
    numberOfRooms: number,
    numberOfStars: number,
    averagePrice:number,
    city: City,
    hotelier: Hotelier
  ) {
    this.id = id;
    this.name = name;
    this.address = address;
    this.email = email;
    this.phone = phone;
    this.image = image;
    this.numberOfRooms = numberOfRooms;
    this.numberOfStars = numberOfStars;
    this.averagePrice = averagePrice
    this.city = city;
    this.hotelier = hotelier;
  }
}

File: superhotelapi/src/main/java/fr/fms/superhotelapi/entites/Hotel.java
package fr.fms.superhotelapi.entites;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor @Builder
public class Hotel implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;
    private String email;
    private String phone;
    private String image;
    private int numberOfRooms;
    private int numberOfStars;

    @ManyToOne
    private City city;

    @ManyToOne
    @JoinColumn(name = "hotelier_id")
    private Hotelier hotelier;

    @Override
    public String toString() {
        return "Hotel{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", address='" + address + '\'' +
                ", email='" + email + '\'' +
                ", phone='" + phone + '\'' +
                ", image='" + image + '\'' +
                ", numberOfRooms=" + numberOfRooms +
                ", numberOfStars=" + numberOfStars +
                ", city=" + city +
                ", hotelier=" + hotelier +
                '}';
    }
}
